src/config.rs

```diff
@@ -173,6 +174,110 @@ impl IndexMut<usize> for ColorList {
     }
 }
 
+/// VisulBellAnimations are modeled after a subset of CSS transitions and Robert
+/// Penner's Easing Functions.
+#[derive(Clone, Copy, Debug, Deserialize)]
+pub enum VisualBellAnimation {
+    Ease,          // CSS
+    EaseOut,       // CSS
+    EaseOutSine,   // Penner
+    EaseOutQuad,   // Penner
+    EaseOutCubic,  // Penner
+    EaseOutQuart,  // Penner
+    EaseOutQuint,  // Penner
+    EaseOutExpo,   // Penner
+    EaseOutCirc,   // Penner
+    Linear,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct VisualBellConfig {
+    /// Visual bell animation function
+    #[serde(default="default_visual_bell_animation")]
+    animation: VisualBellAnimation,
+
+    /// Visual bell duration in milliseconds
+    #[serde(default="default_visual_bell_duration")]
+    duration: u16,
+}
+
+fn default_visual_bell_animation() -> VisualBellAnimation {
+    VisualBellAnimation::EaseOutExpo
+}
+
+fn default_visual_bell_duration() -> u16 {
+    150
+}
+
+impl VisualBellConfig {
+    /// Visual bell animation
+    #[inline]
+    pub fn animation(&self) -> VisualBellAnimation {
+        self.animation
+    }
+
+    /// Visual bell duration in milliseconds
+    #[inline]
+    pub fn duration(&self) -> Duration {
+        Duration::from_millis(self.duration as u64)
+    }
+}
+
+impl Default for VisualBellConfig {
+    fn default() -> VisualBellConfig {
+        VisualBellConfig {
+            animation: default_visual_bell_animation(),
+            duration: default_visual_bell_duration(),
+        }
+    }
+}
+
+impl From<u16> for VisualBellConfig {
+    fn from(duration: u16) -> VisualBellConfig {
+        VisualBellConfig {
+            animation: default_visual_bell_animation(),
+            duration: duration,
+        }
+    }
+}
+
+fn u16_or_struct<T, D>(d: D) -> ::std::result::Result<T, D::Error>
+    where T: de::Deserialize + From<u16>,
+          D: de::Deserializer
+{
+    use std::marker::PhantomData;
+
+    struct U16OrStruct<T>(PhantomData<T>);
+
+    impl<T> de::Visitor for U16OrStruct<T>
+        where T: de::Deserialize + From<u16>
+    {
+        type Value = T;
+
+        fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+            formatter.write_str("u16 or map")
+        }
+
+        fn visit_i64<E>(self, value: i64) -> ::std::result::Result<T, E>
```

Comment:
Why not implement visit_u64 instead? That would at least make sure a negative value cannot be passed.